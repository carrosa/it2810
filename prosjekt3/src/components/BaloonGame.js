import React from 'react';
import {Text, View, StyleSheet, Dimensions, Animated, TouchableOpacity} from 'react-native';
import {createStackNavigator} from 'react-navigation';
import {black} from 'react-native-paper/src/styles/colors';
import {Audio} from "expo"


let styles =
    {
      button: {
        position: "absolute",
        height: 100,
        width: 100
      }
    };


export default class BaloonGame extends React.Component {

  // This generates 5 baloons as defined by the styles
  render() {
    return (
        <View style = {{backgroundColor:"black", height: Dimensions.get("window").height}}>
          <TopAnimation style={styles.button}/>
          <TopAnimation style={styles.button}/>
          <TopAnimation style={styles.button}/>
          <TopAnimation style={styles.button}/>
          <TopAnimation style={styles.button}/>
        </View>
    );
  }
}

class TopAnimation extends React.Component {
  state = {
    sleft: new Animated.Value(10),
    topAnim: new Animated.Value(Dimensions.get("window").height),
    baloonColor: ""
  };

  //this doesnt work atm
  GenerateStartPosition() {
    return Dimensions.get("window").height;
  }

  GenerateLeft() {
    return Math.floor(Math.random() * Dimensions.get("window").width) - 50;
  }

  ChooseRandom(array){
    return array[Math.floor(Math.random() * array.length)];
  }
  GenerateColor() {
    let baloonColors = ['red', 'blue', 'yellow', 'orange', 'green', "purple"] 
    return this.ChooseRandom(baloonColors);}

  GenerateSound(){
    let soundOption = [
      require("../sound/pop1.mp3"),
      require("../sound/pop2.mp3"),
      require("../sound/pop3.mp3")
    ]
    return this.ChooseRandom(soundOption);
  }

  componentDidMount() {
    this.RunAnimation();
  }

  async pressHandler() {
    this.RunAnimation();
    pop = new Audio.Sound()
    let soundloc = this.GenerateSound();
    try {
      await pop.loadAsync(this.GenerateSound());
      await pop.playAsync();
      await pop.setPositionAsync(0);
      // Your sound is playing!
    } catch (error) {
      // An error occurred!
    }
  }

  RunAnimation() {
    let newLeft = this.GenerateLeft();
    let newColor = this.GenerateColor();
    this.setState({sleft: newLeft});
    this.setState({baloonColor: newColor});
    this.state.topAnim.setValue(this.GenerateStartPosition());
    Animated.loop(
        Animated.timing(
            this.state.topAnim,
            {
              toValue: 0,
              duration: 10000
            })).start();
  }

  render() {
    return (
        <Animated.View style={{...this.props.style, top: this.state.topAnim, left: this.state.sleft}}>
          {this.props.children}
          <TouchableOpacity
          style={
            this.props.style
          }
              onPress={() => {
                this.pressHandler();
              }}
          >
            <View
                style={{
                  ...this.props.style,
                  backgroundColor: this.GenerateColor(),
                  borderRadius: 50
                }}
            >
            </View>
          </TouchableOpacity>
        </Animated.View>
    );

  }
}

