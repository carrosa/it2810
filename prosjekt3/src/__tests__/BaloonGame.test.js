import React from 'react';
import renderer from 'react-test-renderer';
import BaloonGame from '../components/BaloonGame';

/*
 * Må bruke fake timers pga Expo.Animated.timing
 * bruker en timer som jest ikke skjønner seg på
 * Dette fikser det
 */
jest.useFakeTimers();

// Test bygge BaloonGame snapshot
test('<BaloonGame /> renders correctly', () => {
  const tree = renderer.create(<BaloonGame/>).toJSON();
  expect(tree).toMatchSnapshot();
});

/*
* Ikke noe mer logikk å teste her
* bare å teste animasjoner og sånn
* tenker at det ikke er nødvendig
*
* Trenger heller ikke teste at TopAnimation
* redrer riktig, for det gjør den hvis rendring av BaloonGame
* ikke failer.
*/