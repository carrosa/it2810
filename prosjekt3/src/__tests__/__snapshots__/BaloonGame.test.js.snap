// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaloonGame /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "height": 1334,
    }
  }
>
  <View
    style={
      Object {
        "height": 100,
        "left": 221,
        "position": "absolute",
        "top": 1334,
        "width": 100,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100,
          "opacity": 1,
          "position": "absolute",
          "width": 100,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "blue",
            "borderRadius": 50,
            "height": 100,
            "position": "absolute",
            "width": 100,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "height": 100,
        "left": 164,
        "position": "absolute",
        "top": 1334,
        "width": 100,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100,
          "opacity": 1,
          "position": "absolute",
          "width": 100,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "green",
            "borderRadius": 50,
            "height": 100,
            "position": "absolute",
            "width": 100,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "height": 100,
        "left": 144,
        "position": "absolute",
        "top": 1334,
        "width": 100,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100,
          "opacity": 1,
          "position": "absolute",
          "width": 100,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "orange",
            "borderRadius": 50,
            "height": 100,
            "position": "absolute",
            "width": 100,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "height": 100,
        "left": 664,
        "position": "absolute",
        "top": 1334,
        "width": 100,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100,
          "opacity": 1,
          "position": "absolute",
          "width": 100,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "orange",
            "borderRadius": 50,
            "height": 100,
            "position": "absolute",
            "width": 100,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "height": 100,
        "left": 11,
        "position": "absolute",
        "top": 1334,
        "width": 100,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100,
          "opacity": 1,
          "position": "absolute",
          "width": 100,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 50,
            "height": 100,
            "position": "absolute",
            "width": 100,
          }
        }
      />
    </View>
  </View>
</View>
`;
